scripts/kubestellar

```diff
@@ -39,9 +39,93 @@ cleanup=0
 espw_name="espw"
 log_folder=$(pwd)/kubestellar-logs
 local_kcp=true
+imws="root:imw1"
+wmws="root:wmw1"
 
 SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
+
+function echoerr() {
+   echo "ERROR: $1" >&2
+}
+
+
+function ensure_imw() {
+    while IFS=':' read -ra items; do
+        for i in "${items[@]}"; do
+            if [ "$i" == "" ] ; then
+                break
+            fi
+            if ! kubectl ws "$i" &> /dev/null ; then #
+                if ! kubectl ws create "$i" --enter &> /dev/null ; then
+                    echoerr 'unable to create "$i"!'
+                    break
+                fi
+            fi
+        done
+    done <<< "$1"
+    if ! kubectl get apibinding "edge.kubestellar.io" &> /dev/null; then
+    	kubectl kcp bind apiexport root:espw:edge.kubestellar.io
+    fi
+}
+
+function ensure_imws() {
+    initial_ws=$(kubectl ws . --short)
+    while IFS=',' read -ra items; do
+        for i in "${items[@]}"; do
+            if [ "$i" == "" ] ; then
+                continue
+            fi
+            echo "Ensuring INV: $i"
+            ensure_imw "$i"
+            kubectl ws use $initial_ws &> /dev/null # go back to the initial state
+        done
+    done <<< "$1"
+    kubectl ws use $initial_ws &> /dev/null
+}
+
+
+function ensure_wmw() {
+    while IFS=':' read -ra items; do
+        len=${#items[@]}
+        if [[ ${len} -eq 0 ]]; then
+            break # nothing to do
+        fi
+        # separate the last item
+        last_item=${items[$len-1]}
+        unset items[$len-1]
+        for i in "${items[@]}"; do
+            if [ "$i" == "" ] ; then
```

Comment:
What if `"$last_item" == ""` ?

Reply:
@mike since we are working with a full path line "root:1:2:last"
Then all pieces except the last one can be `ws use` if the exist or `ws create --enter` if they don't
But the last piece must be either "ws use" if it exist or `ks ensure-wmw` if it does not exists

Reply:
Why the special case for empty string?